pkg/gardenlet/controller/managedseed/managedseed_actuator.go

```diff
@@ -257,23 +282,23 @@ func (a *actuator) ensureSeedDeleted(ctx context.Context, managedSeed *seedmanag
 }
 
 func (a *actuator) deployGardenlet(ctx context.Context, shootClient kubernetes.Interface, managedSeed *seedmanagementv1alpha1.ManagedSeed, shoot *gardencorev1beta1.Shoot) error {
-	// Decode gardenlet configuration
 	gardenletConfig, err := helper.DecodeGardenletConfiguration(&managedSeed.Spec.Gardenlet.Config, false)
 	if err != nil {
 		return err
 	}
 
-	// Check seed spec
 	if err := a.checkSeedSpec(ctx, &gardenletConfig.SeedConfig.SeedTemplate.Spec, shoot); err != nil {
 		return err
 	}
 
-	// Create or update seed secrets
 	if err := a.createOrUpdateSeedSecrets(ctx, &gardenletConfig.SeedConfig.SeedTemplate.Spec, managedSeed, shoot); err != nil {
 		return err
 	}
 
-	// Prepare gardenlet chart values
+	if err := a.createSNIIngressNamespace(ctx, shootClient, *gardenletConfig.SNI.Ingress.Namespace); err != nil {
```

Comment:
Shouldn't we check `gardenletConfig.SNI.Ingress.Namespace` is non-nil before dereferencing it? Actually, `SNI` and `Ingress` are also pointers and should be checked as well I believe. Perhaps you could introduce a helper function to properly get this namespace?

It's true that the resource has defaults to always set this. However, these defaults are only applied when an actual resource is created, but not when the resource is deserialized as part of a template - like in this case. So all of these pointers could well be nil I believe.